Assets/Scripts/Upgrades/Game Logic/HandleCoinsUpgrade.cs

```diff
@@ -0,0 +1,158 @@
+            price = (int)price;            
+            Message.Send(new PriceUpdate(price, id));
+            Message.Send(new SpriteSpawn(id));
+        }
+    }
+
+    void coinspersec(Upgrade msg)
+    {
+        if (msg.id == 15 && msg.id== id)
+        {
+            
+            amount = 0.1;
+                if (upgrades == 0)
+                {
+                    InvokeRepeating("coins", 1f, 1f);
+                }
+                upgrades += amount;
+            
+        }
+        else if (msg.id == 100 && msg.id==id)
+        {
+            amount = 0.5;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 500 && msg.id == id)
+        {
+            amount = 4;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 3000 && msg.id == id)
+        {
+            amount = 10;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 10000 && msg.id == id)
+        {
+            amount = 40;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 40000 && msg.id == id)
+        {
+            amount = 100;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 200000 && msg.id == id)
+        {
+            amount = 400;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 4000 && msg.id == id)
+        {
+            amount = 6666;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 5000 && msg.id == id)
+        {
+            amount = 98765;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 6000 && msg.id == id)
+        {
+            amount = 999999;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
+        else if (msg.id == 7000 && msg.id == id)
+        {
+            amount = 10000000;
+            if (upgrades == 0)
+            {
+                InvokeRepeating("coins", 1f, 1f);
+            }
+            upgrades += amount;
+        }
```

Comment:
make the amount variable a public variable. that way you can set it in the inspector and not have it hard coded like it is here.